Remove cargo.lock; Update to reqwest 0.7

ParadoxSpiral opened the pull request:
Cargo.lock should only be included if this was a binary.

The `IntoUrl` trait will soon be deprecated according to release notes.
I didn't bother to handle the parse error more, because it should only indicate a bug in ddg, is this okay?

Aaronepower commented:
Thanks for your PR! Could you link to the discussion of `IntoUrl`s deprecation? My one hesitation is that this _technically_ a breaking change, and removes some nice ergonomics around `Query`. Of course I think most of this will be resolved when `TryFrom` is stabilised and we can use `http::Uri` instead.

Aaronepower commented on src/query.rs:
```diff
@@ -56,7 +56,25 @@ impl<'a> Query<'a> {
     /// Execute the request and parses it into a `DdgResponse` struct.
     pub fn execute(self) -> Result<Response, Error> {
-        Ok(serde_json::from_reader(reqwest::get(self)?)?)
+        Ok(reqwest::get(self.into_url().unwrap())?.json()?)
```
I don't think it should unwrap, I think it should be `Ok(reqwest::get(self.into_url()?)?.json()?)`. As it is a possibility that the user submits a weird a query and then their entire application crashes. This will probably require an additional enum variant and `From` implementation.

ParadoxSpiral commented:
>Could you link to the discussion of IntoUrls deprecation?

https://docs.rs/reqwest/0.7.2/reqwest/trait.IntoUrl.html
>This trait is "sealed", such that only types within reqwest can implement it. The reason is that it will eventually be deprecated and removed, when std::convert::TryFrom is stabilized.

So another option is to wait for `TryFrom` stabilization, and then switch to that.

Added an `Url` variant to `Error`.

Aaronepower commented:
Okay, this looks good to me. Thank you!
